modules/ce_check.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+from argparse import ArgumentParser
+import datetime
+import time
+import json
+from argo_ams_library import ArgoMessagingService, AmsException
+from NagiosResponse import NagiosResponse
+
+
+def main():
+    TIMEOUT = 18
+
+    parser = ArgumentParser(description="Nagios probe for monitoring the compute engine's flow.")
+    parser.add_argument('-H', dest='host', type=str, default='msg-devel.argo.grnet.gr', help='FQDN of AMS Service')
+    parser.add_argument('--token', type=str, required=True, help='Given token')
+    parser.add_argument('--project', type=str, required=True, help='Project registered in AMS Service')
+    parser.add_argument('--push_topic', type=str, default='create_data', help='Given topic')
+    parser.add_argument('--push_subscription', type=str, default='create_data_sub', help='Push_Subscription name')
+
+    parser.add_argument('--pull_subscription', type=str, default='retrieve_data_sub', help='Push_Subscription name')
+    parser.add_argument('-t', dest='timeout', type=int, default=TIMEOUT, help='Timeout')
+    cmd_options = parser.parse_args()
+
+    run_timestamp = str(datetime.datetime.now())
+
+    nagios = NagiosResponse("System Dataflow at " + run_timestamp + " completed successfully.")
+    ams = ArgoMessagingService(endpoint=cmd_options.host, token=cmd_options.token, project=cmd_options.project)
+    try:
+        # For both subscriptions move their offset to max
+        move_sub_offset_to_max(ams, cmd_options.push_subscription)
+        move_sub_offset_to_max(ams, cmd_options.pull_subscription)
+
+        # publish a message with the current timestamp as its content
+        req_data = {'message': run_timestamp, 'errors': []}
+        d1 = {'data': json.dumps(req_data), 'attributes': {}}
+        print(d1)
+        r = ams.publish(cmd_options.push_topic, d1)
+        print(r)
+        start = time.time()
+        no_resp = True
+        while no_resp:
+            end = time.time()
+            # check if the systsem has written to the retrieve topic
+            resp = ams.pull_sub(cmd_options.pull_subscription)
```

Comment:
Can you pass timeout argument here so that network call does not hang in case of AMS problems? That timeout should be different than this "round-trip timeout" for the probe logic.

Something like `ams.pull_sub(cmd_options.pull_subscription, timeout=cmd_options.global_timeout)`